subprojects/plugins/src/main/java/org/gradle/api/plugins/jvm/JvmEcosystemAttributesDetails.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright 2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.gradle.api.plugins.jvm;
+
+import org.gradle.api.Incubating;
+import org.gradle.api.artifacts.Configuration;
+import org.gradle.api.tasks.SourceSet;
+
+/**
+ * Allows configuration of attributes used for JVM related components.
+ * This can be used both on the producer side, to explain what it provides,
+ * or on the consumer side, to express requirements.
+ *
+ * @since 6.6
+ */
+@Incubating
+public
+interface JvmEcosystemAttributesDetails {
```

Comment:
If we want these APIs to help beyond the JVM world supported by Gradle, I believe there should be a method allowing to add arbitrary attribute.
Is there a reason you did not add such a method?

Reply:
> If we want these APIs to help beyond the JVM world supported by Gradle

It's not the goal of this PR and this is actually in the name of this interface :) There wasn't any use case that I analyzed which required a custom attribute, so I didn't add any way to do this yet. And because it's not a blocker (you can still tweak the attributes after), I didn't add any.